packages/jazz-tools/src/implementation/WebSocketPeerWithReconnection.ts

```diff
@@ -0,0 +1,106 @@
+import { Peer } from "cojson";
+import { createWebSocketPeer } from "cojson-transport-ws";
+
+export class WebSocketPeerWithReconnection {
```

Comment:
Refactored the webSocketPeerWithReconnection function into a class because it's easier to manage the internal state this way.